peripherals/timer/timer_basic.yaml

```diff
@@ -34,27 +34,5 @@ TIMER0:
   SWEVG:
     UPG:
       Update: [1, "Re-initializes the timer counter and generates an update of the registers."]
-
-"TIMER[1-9],TIMER1[0-9]":
-  CTL0:
-    ARSE:
-      _derivedFrom: "TIMER0.CTL0.ARSE.ARSE"
-    UPS:
-      _derivedFrom: "TIMER0.CTL0.UPS.UPS"
-    UPDIS:
-      _derivedFrom: "TIMER0.CTL0.UPDIS.UPDIS"
-    CEN:
-      _derivedFrom: "TIMER0.CTL0.CEN.CEN"
-  DMAINTEN:
-    UPIE:
-      _derivedFrom: "TIMER0.DMAINTEN.UPIE.UPIE"
-  INTF:
-    UPIF:
-      _derivedFrom: "TIMER0.INTF.UPIF.UPIF"
-  SWEVG:
-    UPG:
-      _derivedFrom: "TIMER0.SWEVG.UPG.UPG"
```

Comment:
Doesn't removing all these `_derivedFrom` fixes result in the fields having different types? That's going to break the HAL crate, and mean a lot more code in the PAC.

Reply:
The codegen gave the same types for the identical timers (timer1/2/3/4 for instance), but the type itself is different between different timers indeed :/

Reply:
I get issues when I try to use the derived peripherals:
```
TIMER0:
  CTL0:
    CEN:
      Disabled: [0, "Counter disabled"]
      Enabled: [1, "Counter enabled"]

"TIMER[1-9],TIMER1[0-9]":
  CTL0:
    CEN:
      _derivedFrom: "TIMER0.CTL0.CEN.CEN"
```
If I try to use cen here for instance I get:
```
error[E0599]: no method named `disabled` found for struct `BitWriterRaw<'_, u32, gd32e1::gd32e103::timer1::ctl0::CTL0_SPEC, CEN_A, BitM, 0>` in the current scope
   --> src/timer.rs:361:55
    |
361 |                 self.timer.ctl0.modify(|_, w| w.cen().disabled());
    |                                                       ^^^^^^^^ method not found in `BitWriterRaw<'_, u32, gd32e1::gd32e103::timer1::ctl0::CTL0_SPEC, CEN_A, BitM, 0>`

```

Reply:
I tried compiling the gd32f1x0-hal library on this repo's main branch and I got the same error :thinking:

Reply:
Hmm, I guess I'm going to need to bisect then. I probably won't have time for a few days though. If you figure out what's going on in the meantime let me know.

If it would be helpful I'm happy to merge everything except for the timer changes for now.

Reply:
I think this is a change that happened with the new svd2rust. During code generation the enum type and reader type are just linked to the original one, but the writer is put as a new type. The issue is either that the writer is defined as a new type or that the writer doesn't have any implementation (it should have the methods to set the register values). I'm trying to look in svd2rust's code to understand how this happened.
I don't need a merge right now, we can try and fix this and then merge :)

Reply:
Found it, it appears to be a bug in svd2rust. This crate in v0.6.0 was using svd2rust 0.24.0 (which have been yanked from the registry), and the bug appeared in svd2rust 0.25.0.
I submitted an issue on their repo: https://github.com/rust-embedded/svd2rust/issues/705